### PR TITLE
Fix: Move defaults for max_parallel in AwsQuantumTaskBatch to AwsDevice

### PR DESCRIPTION
**Pull request by licedric**
*Issue #, if available:*

*Description of changes:*
Move the defaults for the `max_parallel` keyword for task batches to `AwsDevice`.
This will allow setting device-dependent defaults, like we currently do for `shots`.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/aws/aws_device.py [hunk 1]
```diff
@@ -201,7 +203,7 @@ def run_batch(
             task_specifications,
             s3_destination_folder,
             shots if shots is not None else self._default_shots,
-            max_parallel=max_parallel,
+            max_parallel=max_parallel if max_parallel is not None else self._default_max_parallel,
```
Comment (speller26) [inline src/braket/aws/aws_device.py:206]: Change the docstring to reflect that this is now an optional int

File: src/braket/aws/aws_device.py [hunk 2]
```diff
@@ -296,6 +298,10 @@ def _default_shots(self):
             AwsDevice.DEFAULT_SHOTS_QPU if "qpu" in self.arn else AwsDevice.DEFAULT_SHOTS_SIMULATOR
         )
 
+    @property
+    def _default_max_parallel(self):
+        return AwsDevice.DEFAULT_MAX_PARALLEL
```
Comment (speller26) [inline src/braket/aws/aws_device.py:303]: Just making sure, this is to future-proof for device-specific defaults, right?

**Review by speller26 — APPROVED** [review]
:shipit: with minor comment

File: src/braket/aws/aws_device.py (hunk 2 above)
Comment (licedric) [reply src/braket/aws/aws_device.py:303]: Yup.

**Pull request merged**